src/filesystem/implementations/gcs.h

```diff
@@ -363,7 +364,8 @@ GCSFileSystem::ReadTextFile(const std::string& path, std::string* contents)
 
 Status
 GCSFileSystem::LocalizePath(
-    const std::string& path, std::shared_ptr<LocalizedPath>* localized)
+    const std::string& path, const bool recursive, const std::string& mount_dir,
```

Comment:
Are you going to implement the new API in other cloud file system in this PR as well? If not, should return error if `!mount_dir.empty()`

Reply:
Yes, I wanted to first get an agreement on the implementation for s3. I'll populate it to GCS and AS in the next commit for easier review